src/content/app/entity-viewer/variant-view/transcript-consequences/transcript-variant-protein/TranscriptVariantProtein.tsx

```diff
@@ -0,0 +1,360 @@
+    alleleType,
+    variantStart,
+    variantEnd,
+    consequences
+  } = props;
+  /**
+   * - uncertain
+   * - number
+   * -
+   */
+
+  // For protein sequences, only treat in-frame insertions as insertions
+  // An insertion that has caused a frame shift will be treated differently (with a consequence being marked as "uncertain").
+  const isInframeInsertion = consequences.some(
+    (conseq) => conseq.value === 'inframe_insertion'
+  );
+
+  let changedSequence: React.ReactNode; // can show modified amino acids, number of amino acids, or a special label such as "uncertain" depending on situation
+  let modificationTypeLabel: React.ReactNode = null;
+
+  if (!variantSequence || variantSequence.match(/X/i)) {
+    // A sequence containing the letter X is interpreted as an unknown sequence.
+    // When no sequence is provided, the result is also unknown
+    changedSequence = (
+      <div className={styles.changedSequenceBlock}>uncertain</div>
+    );
+  } else if (
+    variantSequence.length > MAX_REFERENCE_ALLELE_DISPLAY_LENGTH &&
+    alleleType !== 'deletion'
+  ) {
+    /* show the number of inserted / changed amino acids */
+    changedSequence = (
+      <div className={styles.changedSequenceBlock}>
+        {variantSequence.length}
+      </div>
+    );
+  } else if (alleleType === 'deletion') {
+    /* deletion shows the number of deleted amino acids */
+    changedSequence = (
+      <div className={styles.changedSequenceBlock}>
+        {variantEnd - variantStart + 1}
+      </div>
+    );
+  } else {
+    const letterBlocks = variantSequence
+      .split('')
+      .map((letter, index) => (
+        <SequenceLetterBlock
+          key={index}
+          className={classNames(styles.letter, styles.altAlleleLetter)}
+          letter={letter}
+        />
+      ));
+    changedSequence = <div>{letterBlocks}</div>;
+  }
+
+  if (alleleType === 'deletion') {
+    modificationTypeLabel = <span>deleted</span>;
+  } else if (isInframeInsertion) {
+    modificationTypeLabel = <span>inserted</span>;
+  }
+
+  const numLettersToSequenceMid = Math.floor(
+    DISPLAYED_REFERENCE_SEQUENCE_LENGTH / 2
+  );
+  let leftOffset =
+    SEQUENCE_LETTER_WIDTH * numLettersToSequenceMid +
+    numLettersToSequenceMid +
+    SEQUENCE_LETTER_WIDTH / 2;
+  const variantLength = variantEnd - variantStart - 1;
+  if (variantLength % 2 === 0) {
+    leftOffset -= SEQUENCE_LETTER_WIDTH / 2;
+  }
+
+  const componentStyles = {
+    marginLeft: `${leftOffset}px`
+  };
+
+  return (
+    <div className={styles.changedSequenceContainer} style={componentStyles}>
+      <AltAlleleArrow direction={isInframeInsertion ? 'up' : 'down'} />
+      <span className={styles.proteinImpactLabel}>Protein impact</span>
+      {changedSequence}
+      {modificationTypeLabel && (
+        <span className={styles.modificationTypeLabel}>
+          {modificationTypeLabel}
+        </span>
+      )}
+    </div>
+  );
+};
+
+// FIXME: extract in a reusable component
```

Comment:
The appropriate time to do this is in https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2494